include/counter.h

```diff
@@ -256,11 +256,12 @@ static inline u32_t _impl_counter_read(struct device *dev)
 }
 
 /**
- * @brief Set an alarm on a channel.
+ * @brief Set a single shot alarm on a channel.
  *
- * In case of absolute request, maximal value that can be set is equal to
- * top value set by @ref counter_set_top_value call or default, maximal one. In
- * case of relative request, maximal value can be retrieved using
+ * After expiration alarm can be set again, disabling is not needed. In case of
+ * absolute request, maximal value that can be set is equal to top value set by
+ * @ref counter_set_top_value call or default, maximal one. In case of relative
+ * request, maximal value can be retrieved using
```

Comment:
I don't think it's necessary to repeat the description of the maximum delay; that's already documented at `ticks` in `struct counter_alarm_cfg` (with wording that's a little less ambiguous than what's here).

I would like to see clear documentation on whether the callback is permitted to reschedule the alarm, and if there are any constraints on whether the callback can schedule or cancel alarms on other channels within the same counter.  Absence of this information has caused problems in the past (see #12515) and to ensure cross-platform consistency we need to make the expectations clear for both implementers and users.

Reply:
i updated the comment. I've also added a note that `set_channel_alarm` and `disable_channel_alarm` are not thread safe.